expression/builtin/time.go

```diff
@@ -309,6 +309,26 @@ func builtinCurrentDate(args []interface{}, ctx map[interface{}]interface{}) (in
 		Type: mysql.TypeDate, Fsp: 0}, nil
 }
 
+// See https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_curtime
+func builtinCurrentTime(args []interface{}, ctx map[interface{}]interface{}) (interface{}, error) {
```

Comment:
Please use mysql.Duration instead of mysql.Time, or first you can use the mysql.Time to handle the argument, then convert it to mysql.Duration. 

mysql.Time is not for duration, only for datetime, timestamp and date.